_source/_posts/2024-04-30-express-universal-logout.md

```diff
@@ -183,22 +185,72 @@ universalLogoutRoute.post('/global-token-revocation', async (req, res) => {
   if (!req.body) {
     res.status(400);
   }
-	
-  // Find the user by email linked to the org id associated with the API key provided
-  const domainOrgId = req['user']['id']
+
+  // Find the user
   const newRequest:IRequestSchema = req.body;
   const { email } = newRequest.sub_id;
   const user = await prisma.user.findFirst({
     where: {
-      email: email,
-      org: { id: domainOrgId },
+      email: email
     },
   });
+  return res.sendStatus(httpStatus);
 
-  // 404 User not found 
+  // 404 User not found
   if (!user) {
     res.sendStatus(404);
   }
+
+});
+
+universalLogoutRoute.use((err,req,res,next) => {
+  if(err){
+    return res.sendStatus(404)
+  }
+})
+```
+The apps/api/src/universalLogout.ts file now looks like the following:
+
+```ts
+import { Router } from 'express';
+export const universalLogoutRoute = Router();
+import { PrismaClient } from '@prisma/client';
+const prisma = new PrismaClient();
+
+interface IRequestSchema {
+  'sub_id': {format:string; email: string};
+}
+universalLogoutRoute.post('/global-token-revocation', async (req, res) => {
+  // 204 When the request is successful
+  const httpStatus = 204;
+
+  // 400 If the request is malformed
+  if (!req.body) {
+    res.status(400);
+  }
+
+  // Find the user
+  const newRequest:IRequestSchema = req.body;
+  const { email } = newRequest.sub_id;
+  const user = await prisma.user.findFirst({
+    where: {
+      email: email
+    },
+  });
+  return res.sendStatus(httpStatus);
+
+  // 404 User not found
+  if (!user) {
+    res.sendStatus(404);
+  }
+
+});
+
+universalLogoutRoute.use((err,req,res,next) => {
+  if(err){
```

Comment:
What's an error and why are they all 404?

Reply:
wow I'm definitely missing here too

`return res.sendStatus(httpStatus);
});`